File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    let answer = 0
    const memo = {}
    
    for (let first = 0; first < nums.length - 2; first++) {
        for (let second = first + 1; second < nums.length - 1; second++ ) {
            for (let third = second + 1; third < nums.length; third++) {
                const sum = nums[first] + nums[second] + nums[third]
                if (memo[sum] || isPrime(sum)) {
                    answer += 1
                    if (memo[sum] === undefined) {
                        memo[sum] = true
                    }
                }
            }
        }
    }
    return answer
}

function isPrime(num) {
    if (num === 2) return true
    if (num % 2 === 0) return false
    
    for (let i = 3; i <= Math.sqrt(num); i += 2) {
        if (num % i === 0) {
            return false
        }
    }
    return true
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
import math

def is_prime(num):
    if num == 2:
        return True
    if num % 2 == 0 or num < 2:
        return False
    
    for i in range(3, int(math.sqrt(num)) + 1, 2):
        if num % i == 0:
            return False
    return True

def solution(nums):
    answer = 0
    memo = {}
    
    for first in range(len(nums) - 2):
        for second in range(first + 1, len(nums) - 1):
            for third in range(second + 1, len(nums)):
                current_sum = nums[first] + nums[second] + nums[third]
                if current_sum in memo or is_prime(current_sum):
                    answer += 1
                    if current_sum not in memo:
                        memo[current_sum] = True
    return answer